lib/book.rb

```diff
@@ -0,0 +1,17 @@
+class Book
+
+  attr_reader :title, :author, :publication_year
+
+  def initialize(book_info)
+    @book_info = book_info
+    @author_first_name = book_info[:author_first_name]
+    @author_last_name = book_info[:author_last_name]
```

Comment:
same here, we don't ask that first/last name get maintained for anything; be careful to only build what we ask for!